plugins/obs-outputs/rtmp-stream.c

```diff
@@ -1228,6 +1229,78 @@ static int try_connect(struct rtmp_stream *stream)
 	return init_send(stream);
 }
 
+static bool build_dbr_interpolation_table(struct rtmp_stream *stream, const char *interpolation_data)
+{
+	json_error_t error;
+	json_t *root = json_loads(interpolation_data, JSON_REJECT_DUPLICATES, &error);
+	if (!root) {
+		warn("loading dbr interpolation table failed: %s", error.text);
+		return false;
+	}
+
+	size_t all_array_size = 0;
+	bool malformed_data = false;
+	for (size_t i = 0; i < MAX_OUTPUT_VIDEO_ENCODERS; i++) {
+		obs_encoder_t *enc = obs_output_get_video_encoder2(stream->output, i);
+		if (!enc)
+			continue;
+
+		const char *name = obs_encoder_get_name(enc);
+		const json_t *array = json_array_get(root, i);
+		if (!array) {
+			warn("missing bitrate interpolation data for "
+			     "encoder '%s'",
+			     name);
+			malformed_data = true;
+			continue;
+		}
+
+		size_t size = json_array_size(array);
+		if (!all_array_size) {
+			all_array_size = size;
+			continue;
+		}
+
+		if (all_array_size != size) {
+			warn("size %zu of bitrate interpolation data for "
+			     "encoder '%s' does not match overall size %zu",
+			     size, name, all_array_size);
```

Comment:
Unwrap and rewrap. Column limit is 120. Generally, logged strings should be unbroken.